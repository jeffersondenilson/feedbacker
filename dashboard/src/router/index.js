import { createRouter, createWebHistory } from 'vue-router'

// route level code-splitting
// this generates a separate chunk (about.[hash].js) for this route
// which is lazy-loaded when the route is visited.

const Home = () => import('../views/Home/index')
const Feedbacks = () => import('../views/Feedbacks/index')
const Credentials = () => import('../views/Credentials/index')

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/credentials',
    name: 'Credentials',
    component: Credentials,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' }
  }
]

const router = createRouter({
  // process.env.BASE_URL
  history: createWebHistory('/'),
  routes
})

export default router
