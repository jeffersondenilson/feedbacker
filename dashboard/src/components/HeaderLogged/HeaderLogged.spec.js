import { createRouter, createWebHistory } from 'vue-router'
import { shallowMount } from '@vue/test-utils'
import HeaderLogged from '.'
import { routes } from '@/router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

const mockStore = { currentUser: {} }
jest.mock('@/hooks/useStore', () => {
  // executa quando o arquivo Ã© importado
  return () => {
    return mockStore
  }
})

describe('<HeaderLogged />', () => {
  it('should render header-logged correctly', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router]
      }
    })
    
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render ... when there\'s no user logged', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router]
      }
    })
    
    const logoutButton = wrapper.find('[data-test=logout-button]')
    expect(logoutButton.text()).toBe('...')
  })

  it('should render username when there\'s user logged', async () => {
    router.push('/')
    await router.isReady()
    mockStore.currentUser.name = 'User Name'
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router]
      }
    })

    const logoutButton = wrapper.find('[data-test=logout-button]')
    expect(logoutButton.text()).toBe('User Name (sair)')
  })
})
