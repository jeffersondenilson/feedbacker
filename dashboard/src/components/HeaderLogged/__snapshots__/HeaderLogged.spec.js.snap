// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> should render header-logged correctly 1`] = `
<div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
  <div class="hidden sm:block w-24 lg:w-36"><img class="w-full" src="" alt="logo"></div>
  <div class="flex">
    <ul class="flex flex-wrap justify-center items-center list-none">
      <li class="px-4 md:px-6 py-2 mr-2 font-bold text-white
          rounded-full cursor-pointer focus:outline-none"> Credenciais </li>
      <li class="px-4 md:px-6 py-2 mr-2 font-bold text-white
          rounded-full cursor-pointer focus:outline-none"> Feedbacks </li>
      <li class="mt-2 px-6 py-2 text-sm md:text-base font-bold bg-white text-brand-main
          rounded-full cursor-pointer focus:outline-none" data-test="logout-button">...</li>
    </ul>
  </div>
</div>
`;
